src/actions.rs

```diff
@@ -86,63 +144,18 @@ impl ActionHandler {
         match result {
             BuildResult::Success(ref x) | BuildResult::Failure(ref x) => {
                 debug!("build - Success");
+
+                self.clear_build_results();
+
                 {
                     let mut results = self.previous_build_results.lock().unwrap();
```

Comment:
Hmm, I think you can hold the lock while you iterate over the results, it doesn't seem better to take the lock on each iteration. However, I still think you should take the lock in `parse_compiler_messages`, there is no reason to pass in `results`.